engine/crates/postgres-types/src/transport.rs

```diff
@@ -31,23 +31,25 @@ impl Column {
 
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
-pub trait Transport {
+pub trait Transport: Send + Sync {
     async fn parameterized_execute(&self, query: &str, params: Vec<Value>) -> crate::Result<i64>;
 
-    async fn parameterized_query<T>(&self, query: &str, params: Vec<Value>) -> crate::Result<Vec<T>>
-    where
-        T: DeserializeOwned + Send;
+    async fn parameterized_query(&self, query: &str, params: Vec<Value>) -> crate::Result<Vec<Value>>;
```

Comment:
I think the map_result function works fine - but you could also have split this trait in two like I did with this `Context` trait here: https://github.com/grafbase/grafbase/blob/main/engine/crates/engine/src/context/ext.rs#L35

You put the `parameterized_execute` function in a `Transport` trait.  That trait will be dyn safe, then you put all the non dyn safe functions on a `TransportExt` trait and [implement that for the dyn safe trait like I did here](https://github.com/grafbase/grafbase/blob/main/engine/crates/engine/src/context/ext.rs#L303C66-L303C66)

Reply:
Me and Julius had a chat and opted to instead return a `Stream<Result<Value, Error>>` from the `Transport` . Felt more adequate to bubble up to the caller any further processing.